core/src/test/java/org/apache/iceberg/io/inmemory/InMemoryFileIOTest.java

```diff
@@ -0,0 +1,196 @@
+import java.util.stream.IntStream;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.io.PositionOutputStream;
+import org.apache.iceberg.io.SeekableInputStream;
+import org.junit.Before;
+import org.junit.Test;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class InMemoryFileIOTest {
+
+  private InMemoryFileIO fileIO;
+  private InMemoryFileStore store;
+
+  @Before
+  public void before() {
+    fileIO = new InMemoryFileIO();
+    store = fileIO.getStore();
+  }
+
+  @Test
+  public void testGetStore() {
+    assertNotNull(store);
+  }
+
+  @Test
+  public void testDeleteFile() {
+    assertFalse(store.exists("file1"));
+
+    // Create the file
+    store.put("file1", ByteBuffer.wrap(new byte[0]));
+    assertTrue(store.exists("file1"));
+
+    // Delete the file
+    fileIO.deleteFile("file1");
+    // Verify that the file has been deleted
+    assertFalse(store.exists("file1"));
+  }
+
+  @Test
+  public void testNewInputFile() throws IOException {
+    String fileName = "file1";
+
+    store.put(fileName, ByteBuffer.wrap("data1".getBytes(UTF_8)));
+
+    InputFile inputFile = fileIO.newInputFile(fileName);
+    assertNotNull(inputFile);
+
+    assertTrue(inputFile.exists());
+    assertEquals(fileName, inputFile.location());
+    assertEquals(5, inputFile.getLength());
+
+    SeekableInputStream inputStream = inputFile.newStream();
+    assertNotNull(inputStream);
+
+    // Test read()
+    assertEquals(0, inputStream.getPos());
+    assertEquals('d', inputStream.read());
+
+    // Test read(byte[], index, len)
+    byte[] dataRead = new byte[3];
+    assertEquals(2, inputStream.read(dataRead, 0, 2));
+    assertEquals("at", new String(dataRead, 0, 2, UTF_8));
+
+    inputStream.close();
+  }
+
+  @Test
+  public void testSeek() throws IOException {
+    String fileName = "file1";
+
+    // Number of rows
+    int numRowsInChunk = 10;
+
+    // Emulate parquet file structure
+    ByteBuffer buffer = ByteBuffer.allocate(1024);
+    // Add magic number
+    buffer.put("par1".getBytes(UTF_8));
+    // Add column chunk 1 for int32
+    IntStream.range(0, numRowsInChunk).forEach(buffer::putInt);
+    // Add column chunk 2 for float64
+    IntStream.range(0, numRowsInChunk).forEach(buffer::putDouble);
+    // Add metadata footer (index to the start of chunks)
+    buffer.putInt(4);
+    buffer.putInt(4 + numRowsInChunk * 4);
+    // Add footer size
+    buffer.putInt(8);
+    // Add magic number
+    buffer.put("par1".getBytes(UTF_8));
+
+    // Put the data in the store
+    buffer.flip();
+    store.put(fileName, buffer);
+
+    // magic number, chunk1, chunk2, footer, footer size, magic number
+    int expectedFileSize = 4 + numRowsInChunk * 4 + numRowsInChunk * 8 + 8 + 4 + 4;
```

Comment:
Nit: for the `4 * numRowsInChunk` and `numRowsInChunk * 8` (and generally for all of the commented magic numbners), could you make named constants for them?

Something like `int expectedFileSize = MAGIC_NUMBER_SIZE + numRowsInChunk * BYTES_IN_INT32 + numRowsInChunk * BYTES_IN_FLOAT64 + FOOTER + SIZE_OF_FOOTER + ....` (or however it is exactly that these numbers work out)?

It's a bit hard to grok on first appearance and I think that would help quite a lot.